pkg/controllers/success_policy_test.go

```diff
@@ -0,0 +1,283 @@
+				SuccessPolicy(&jobset.SuccessPolicy{
+					TargetReplicatedJobs: []string{"test-replicated-job-1", "test-replicated-job-2"},
+				}).Obj(),
+			jobs: []*batchv1.Job{
+				testutils.MakeJob(jobName, ns).
+					JobLabels(map[string]string{jobset.ReplicatedJobNameKey: "test-replicated-job-1"}).
+					Obj(),
+				testutils.MakeJob(jobName, ns).
+					JobLabels(map[string]string{jobset.ReplicatedJobNameKey: "test-replicated-job-3"}).
+					Obj(),
+			},
+			expected: 1,
+		},
+		{
+			name: "all jobs matches the success policy",
+			js: testutils.MakeJobSet(jobSetName, ns).
+				SuccessPolicy(&jobset.SuccessPolicy{
+					TargetReplicatedJobs: []string{"test-replicated-job-1", "test-replicated-job-2"},
+				}).Obj(),
+			jobs: []*batchv1.Job{
+				testutils.MakeJob(jobName, ns).
+					JobLabels(map[string]string{jobset.ReplicatedJobNameKey: "test-replicated-job-1"}).
+					Obj(),
+				testutils.MakeJob(jobName, ns).
+					JobLabels(map[string]string{jobset.ReplicatedJobNameKey: "test-replicated-job-2"}).
+					Obj(),
+			},
+			expected: 2,
+		},
+		{
+			name: "non job matches the success policy",
+			js: testutils.MakeJobSet(jobSetName, ns).
+				SuccessPolicy(&jobset.SuccessPolicy{
+					TargetReplicatedJobs: []string{"test-replicated-job-1", "test-replicated-job-2"},
+				}).Obj(),
+			jobs: []*batchv1.Job{
+				testutils.MakeJob(jobName, ns).
+					JobLabels(map[string]string{jobset.ReplicatedJobNameKey: "test-replicated-job-3"}).
+					Obj(),
+				testutils.MakeJob(jobName, ns).
+					JobLabels(map[string]string{jobset.ReplicatedJobNameKey: "test-replicated-job-4"}).
+					Obj(),
+			},
+			expected: 0,
+		},
+		{
+			name: "empty target replicated jobs, all job matches the success policy",
+			js: testutils.MakeJobSet(jobSetName, ns).
+				SuccessPolicy(&jobset.SuccessPolicy{
+					TargetReplicatedJobs: []string{},
+				}).Obj(),
+			jobs: []*batchv1.Job{
+				testutils.MakeJob(jobName, ns).
+					JobLabels(map[string]string{jobset.ReplicatedJobNameKey: "test-replicated-job-1"}).
+					Obj(),
+				testutils.MakeJob(jobName, ns).
+					JobLabels(map[string]string{jobset.ReplicatedJobNameKey: "test-replicated-job-2"}).
+					Obj(),
+			},
+			expected: 2,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			actual := numJobsMatchingSuccessPolicy(tc.js, tc.jobs)
+			if diff := cmp.Diff(tc.expected, actual); diff != "" {
+				t.Errorf("unexpected finished value (+got/-want): %s", diff)
+			}
+		})
+	}
+}
+
+func TestNumJobsExpectedToSucceed(t *testing.T) {
+	var (
+		jobSetName = "test-jobset"
+		ns         = "default"
+	)
+
+	tests := []struct {
+		name     string
+		js       *jobset.JobSet
+		expected int
+	}{
+		{
+			name: "any single job matching the jobSelector",
+			js: testutils.MakeJobSet(jobSetName, ns).
+				SuccessPolicy(&jobset.SuccessPolicy{
+					Operator: jobset.OperatorAny,
+				}).Obj(),
+			expected: 1,
+		},
+		{
+			name: "all replicated jobs match success policy",
```

Comment:
In this test case, `"test-replicated-job-3"` does not match the success policy, so we should remove it so we are testing the case where all replicatedJobs match the success policy.